### PR TITLE
RD-2151 Added display_name support to deployments view widget

### PR DESCRIPTION
**Pull request by kubama**
Table view with added ID column:
image

System tests: https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/600/

Additional change: proper clean up before deployments view spec.

### CODE REVIEW

File: test/cypress/support/blueprints.ts [hunk 1]
```diff
@@ -24,7 +24,10 @@ const commands = {
         );
     },
     deleteBlueprint: (blueprintId: string, force = false) => {
-        cy.cfyRequest(`/blueprints/${blueprintId}?force=${force}`, 'DELETE');
+        cy.cfyRequest(`/deployments?blueprint_id=${blueprintId}`, 'GET')
+            .then(response => response.body.items.forEach(({ id }: { id: string }) => cy.deleteDeployment(id, force)))
+            .then(() => waitUntilEmpty(`deployments?blueprint_id=${blueprintId}`));
+        cy.cfyRequest(`/blueprints/${blueprintId}?force=${force}`, 'DELETE', null, null, { failOnStatusCode: false });
```
Comment (qooban) [inline test/cypress/support/blueprints.ts:30]: Good idea 👍🏻

File: widgets/common/src/deploymentsView/detailsPane/DrilldownButtons.tsx [hunk 2]
```diff
@@ -43,13 +43,13 @@ const DrilldownButtons: FunctionComponent<DrilldownButtonsProps> = ({ drillDown,
...
                 drillDown={drillDown}
-                deploymentName={deploymentId}
+                deploymentName={deployment.display_name}
                 result={subdeploymentResults.subenvironments}
             />
             <DrilldownButton
                 type="services"
                 drillDown={drillDown}
-                deploymentName={deploymentId}
+                deploymentName={deployment.display_name}
```
Comment (qooban) [inline widgets/common/src/deploymentsView/detailsPane/DrilldownButtons.tsx:52]: You can mayb extract (destructure) `deployment` and get both - `id` and `displayName` fields and use it.

File: widgets/common/src/deploymentsView/detailsPane/header.tsx [hunk 3]
```diff
@@ -26,12 +27,12 @@ const DetailsPaneHeader: FunctionComponent<DetailsPaneHeaderProps> = ({ deployme
...
             maximized: false
         }),
-        [deploymentName]
+        [deployment.id]
     );
 
     return (
         <div className="detailsPaneHeader">
-            <Header>{deploymentName}</Header>
+            <Header>{deployment.display_name}</Header>
```
Comment (qooban) [inline widgets/common/src/deploymentsView/detailsPane/header.tsx:35]: Similar comment to my previous one: You can extract (destructure) `deployment` and get both - `id` and `displayName` fields and use it.

**Review by qooban — COMMENTED** [review]
Looks good. Added some minor comments.

File: widgets/common/src/deploymentsView/detailsPane/DrilldownButtons.tsx (hunk 2 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/detailsPane/DrilldownButtons.tsx]: Done

File: widgets/common/src/deploymentsView/detailsPane/header.tsx (hunk 3 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/detailsPane/header.tsx]: Done

**Review by qooban — APPROVED** [review]

**Comment by Gelio** [comment]
I will review the PR later today (probably closer to the evening)

File: widgets/common/src/deploymentsView/table/columns.tsx [hunk 4]
```diff
@@ -44,10 +45,16 @@ const partialDeploymentsViewColumnDefinitions: Record<
...
         label: ''
     },
+    id: {
+        width: '20px',
+        render(deployment) {
+            return <Stage.Shared.IdPopup id={deployment.id} />;
+        }
+    },
     name: {
         sortFieldName: 'id',
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/table/columns.tsx:55]: Should this not be `display_name`?

**Review by Gelio — DISMISSED** [review]
Looks good to me overall :+1: I just have 1 doubt that I raised :slightly_smiling_face:

**Comment by Gelio** [comment]
The system tests are still failing :worried: Did you run them after doing the additional fixes?

**Review by qooban — APPROVED** [review]

**Comment by kubama** [comment]
> The system tests are still failing 😟 Did you run them after doing the additional fixes?

Yes, I verified locally and they were fine. I'll schedule full system tests build to double check.

File: widgets/common/src/deploymentsView/table/columns.tsx (hunk 4 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/table/columns.tsx]: Definitely - fixed

**Review by Gelio — APPROVED** [review]
Looks good again :+1: Let's just make sure the tests are actually passing :slightly_smiling_face:

**Comment by kubama** [comment]
> Looks good again 👍 Let's just make sure the tests are actually passing 🙂

Looks like system tests fail in the same manner as on the target 6.0.0 branch. I assume these failures are covered by corresponding bugs created as part of the nightly tracking process. Thus, I'm merging this PR.

**Comment by Gelio** [comment]
@kubama 
> Looks like system tests fail in the same manner as on the target 6.0.0 branch. I assume these failures are covered by corresponding bugs created as part of the nightly tracking process. Thus, I'm merging this PR.

There are 3 known failures that are reported RD-2317, but the "Deployments View widget bulk actions should allow to create child deployments on filtered deployments" I did not find to be connected to the default Environments system filter being missing, I thought that was the one you wanted to fix in this PR. I did not report it because I communicated it on Slack https://cloudify.slack.com/archives/G9MN0TDG8/p1621238158005900

Looking at the screenshot from that bug:
image
it does not seem to be connected to RD-2317, at least to me, so looks like it still needs fixing

**Comment by kubama** [comment]
@Gelio
"Deployments View widget bulk actions should allow to create child deployments on filtered deployments" actually had another issue which was occurring on this branch - it was related to test data interference with map feature tests and it was fixed here.
The issue you are referring to is another issue present on 6.0.0. I can work on it, but not necessarily as part of this PR as it is unrelated.

**Comment by Gelio** [comment]
I see, ok, if you have time then go ahead, as I did not report it earlier. I can report it tomorrow when I'll be doing the tracking report or you could do it yourself :slightly_smiling_face:

**Pull request merged**